docs/reference/notion.md

```diff
@@ -0,0 +1,113 @@
+
+## Connecting to Notion
+
+To connect to Notion, the user needs to [create an internal integration](https://www.notion.so/help/create-integrations-with-the-notion-api#create-an-internal-integration) for their Notion workspace, which can access resources using the Internal Integration Secret Token. Configure the Integration with following settings:
+
+1. Users must grant `READ` permission for content, comment and user capabilities for that integration from the Capabilities tab.
+
+2. Users must manually [add the integration as a connection](https://www.notion.so/help/add-and-manage-connections-with-the-api#add-connections-to-pages) to each page and database that needs to be retrieved.
+
+Once completed, use the following parameters to configure the connector.
+
+## Configuration
+
+### Configure Notion connector
+
+Note the following configuration fields:
+
+#### `Notion Secret Key`  (required)
+
+Secret token assigned to your integration, for a particular workspace. Example:
+
+- `zyx-123453-12a2-100a-1123-93fd09d67394`
+
+#### `Databases`  (required)
+
+Comma-separated list of database names to be fetched by the connector. If the value is `*`, connector will fetch all the databases available in the workspace. Example:
+
+- `database1, database2`
+- `*`
+
+#### `Pages`  (required)
+
+Comma-separated list of page names to be fetched by the connector. If the value is `*`, connector will fetch all the pages available in the workspace. Examples:
+
+- `*`
+- `Page1, Page2`
+
+#### `Index Comments`
+
+Toggle to enable fetching and indexing of comments from the Notion workspace for the configured pages, databases and the corresponding child blocks. Default value is `False`.
+
+### Content Extraction
+
+Refer to [content extraction](https://www.elastic.co/guide/en/enterprise-search/current/connectors-content-extraction.html) in the official docs.
+
+## Documents and syncs
+
+The connector syncs the following objects and entities:
+- **Pages**
+    - Includes metadata such as page name, id, last updated time, etc.
+- **Blocks**
+    - Includes metadata such as title, type, id, content (in case of file block), etc.
+- **Databases**
+    - Includes metadata such as name, id, records, size, etc.
+- **Users**
+    - Includes metadata such as name, id, email address, etc.
+- **Comments**
+    - Includes the content and metadata such as id, last updated time, created by, etc.
+
+*NOTE*:
+- Files bigger than 10 MB won't be extracted.
+- Permissions are not synced. **All documents** indexed to an Elastic deployment will be visible to **all users with access** to that Elastic Deployment.
+
+## Sync rules
+
+[Basic sync rules](https://www.elastic.co/guide/en/enterprise-search/current/sync-rules.html#sync-rules-basic) are identical for all connectors and are available by default.
+
+## Advanced Sync Rules
+
+Advanced sync rules are not available for this connector in the present version.
```

Comment:
Will there be a follow-up PR to update this? I believe the design document I'd seen had planned for Advanced Sync Rules